### PR TITLE
Fix a bunch of compiler warnings

### PR DESCRIPTION
**Pull request by greglandrum**
g++ v5.2 and v4.8 now compile most everything without warnings.

### CODE REVIEW

File: Code/GraphMol/MolAlign/O3AAlignMolecules.cpp [hunk 1]
```diff
@@ -1092,7 +1093,7 @@ void SDM::prepareMatchWeightsVect(
     void *data) {
   PRECONDITION(matchVect.size() == weights.size(),
                "matchVect/weights size mismatch");
-  double min;
+  double min = MAX_DOUBLE;
```
Comment (bp-kelley) [inline Code/GraphMol/MolAlign/O3AAlignMolecules.cpp]: perhaps use numeric_limits:max() here.

File: Code/GraphMol/SmilesParse/SmilesWrite.cpp [hunk 2]
```diff
@@ -422,9 +422,9 @@ std::string MolToSmiles(const ROMol &mol, bool doIsomericSmiles, bool doKekule,
     if (canonical) {
       if (tmol->hasProp("_canonicalRankingNumbers")) {
```
Comment (bp-kelley) [inline Code/GraphMol/SmilesParse/SmilesWrite.cpp]: prefer common_properties::_canonicalRankingNumber if possible, should be quicker in general.

**Comment by bp-kelley** [comment]
Feels good in a weird way, doesn't it?

File: Code/GraphMol/MolAlign/O3AAlignMolecules.cpp (hunk 1 above)
Comment (greglandrum) [inline Code/GraphMol/MolAlign/O3AAlignMolecules.cpp]: Yeah, it's in types.h.

File: Code/GraphMol/SmilesParse/SmilesWrite.cpp (hunk 2 above)
Comment (greglandrum) [inline Code/GraphMol/SmilesParse/SmilesWrite.cpp]: Good point. Want to go ahead and just do it ?

**Comment by greglandrum** [comment]
Oh yeah, doing this kind of exercise once in a while is great.

And the unused-variable warnings from types.h were driving me nuts anyway.

**Pull request merged**